### ISSUE TITLE
journalctl -u 'xxx' - add switch to show logs since unit restart

### ISSUE THREAD

**Issue by zabbal** [systemd/systemd#1942]
When debugging misbehaving service which you've got to restart after changing configuration it's really handy to be able to see just the logs from latest run - I mean after running "systemctl restart xxx".

The unit is restarted by systemd so it "knows" when it happened precisely. Would be great if I could use this from within journalctl as well without manually copy-pasting time back and forth. Smth like:
journalctl -u xxx --since-unit-restart

It's in essence similar to journalctl -b but on per-unit, not per-system basis.

**Comment by keszybz** [comment]
For added points, it should be possible to generate a list of restarts, something analogous to --list-boots.

**Comment by towolf** [comment]
I had to make a really hacky bash script to achieve something similar.

Somehow the 'big race' in journal that remains a big problem, also plays into this. 

Journal seems to have no good way of correlating service start and journal timestamps.

```bash
#!/bin/bash
#

[ "${FLOCKER}" != "$0" ] && exec env FLOCKER="$0" flock -en "$0" "$0" "$@" || :

# Timestamp when unit transitioned from inactive to active
since=$(systemctl show -p InactiveExitTimestamp "$1" | cut -f 2 -d '=' | cut -f 2-3 -d' ')
# or one minute if unset
since=${since:-1 min ago}

# Get prefix string that this units logs with: most robust
# https://github.com/systemd/systemd/issues/2913#issuecomment-219702148
id=$(systemctl show -p SyslogIdentifier "$1" | cut -f 2 -d '=')

# Get all raw output from unit since start, only from stdout&stderr
# Considering that backend only logs "bad" stack traces to stderr, this should
# always be relevant
service_trace=$(journalctl -o cat --since "$since" -t "$id")

**Comment by fsateler** [comment]
It should probably make use of `_SYSTEMD_INVOCATION_ID` as introduced by #4067

**Comment by towolf** [comment]
I don't have that field in my journal yet. Version is 229.

**Comment by dvaerum** [comment]
This feature would be very useful, just to draw some attention to this request 😅

**Comment by sedrubal** [comment]
> It should probably make use of `_SYSTEMD_INVOCATION_ID` as introduced by #4067

Is there an easy way to get the invocation id of a unit? Something like `--list-boots`?

**Comment by laurivosandi** [comment]
Systemd never ceases to amaze how obscure overengineered features get prioritized over very basic usability features

```
journalctl -u foobar.service  _SYSTEMD_INVOCATION_ID="$(systemctl show -p InvocationID --value foobar.service)"
```

**Comment by SjonHortensius** [comment]
@laurivosandi you are free to send a PR to implement this, right? Also, what constitutes a basic feature versus an obscure feature seems pretty subjective to me.

Your oneliner is pretty nice by the way, I'll make an alias out of it

**Comment by n0099** [comment]
> journalctl -u foobar.service  _SYSTEMD_INVOCATION_ID="$(systemctl show -p InvocationID --value foobar.service)"

`-u foobar.service` is unnecessary as `_SYSTEMD_INVOCATION_ID` is uniquely bind to a single unit.

> This should be the default for `journalctl -u foobar`

As #5317 has been fixed by #32448, since systemd 257.1 you may use `journalctl -Iu foobar.service` instead.
